aggregator/internal/pkg/aggregator.go

```diff
@@ -193,44 +210,65 @@ func (agg *Aggregator) sendAggregatedResponseToContract(blsAggServiceResp blsagg
 		"taskIndex", blsAggServiceResp.TaskIndex,
 	)
 
-	agg.tasksMutex.Lock()
-	batchMerkleRoot := agg.tasks[blsAggServiceResp.TaskIndex]
-	agg.tasksMutex.Unlock()
+	agg.batchesRootByIdxMutex.Lock()
+	batchMerkleRoot := agg.batchesRootByIdx[blsAggServiceResp.TaskIndex]
+	agg.batchesRootByIdxMutex.Unlock()
 
 	_, err := agg.avsWriter.SendAggregatedResponse(context.Background(), batchMerkleRoot, nonSignerStakesAndSignature)
 	if err != nil {
 		agg.logger.Error("Aggregator failed to respond to task", "err", err)
 	}
 }
 
+// BatchIndex
```

Comment:
small nit, but I think this is not needed

Reply:
I missed it, thanks